ui_framework/src/components/tabs/_tabs.scss

```diff
@@ -0,0 +1,52 @@
+.kuiTabs {
```

Comment:
A little too much on hover. But you've got the right idea by wanting some animation here. The below code simplifies your markup slightly and adds an animation on the selection, rather than the hover. I also went ahead and added our standard K7 patterns for accessibility (both hover underlines and focus backgrounds).

I did also change out the tab color to a color. Think it was a little too plain with the black.

![](https://d3vv6lp55qjaqc.cloudfront.net/items/0a2E1q1R2H0r2i2X1a3J/Screen%20Recording%202017-08-01%20at%2009.28%20PM.gif?X-CloudApp-Visitor-Id=59773&v=5adae46c)

```
.kuiTabs {
  display: flex;
  border-bottom: $kuiBorderThin;
}

/**
 * 1. Move the tab down so that its bottom stripe will cover the container's bottom border.
 */
.kuiTab {
  @include kuiFontSize;

  position: relative;
  cursor: pointer;
  padding: ($kuiSize - $kuiSizeXS) $kuiSize;
  color: $kuiColorDarkShade;
  background-color: $kuiColorEmptyShade;
  transition: all $kuiAnimSpeedNormal $kuiAnimSlightResistance;

  &:hover:not(.kuiTab-isSelected) {
    color: $kuiTextColor;
    text-decoration: underline;
  }

  &:focus {
    background-color: $kuiFocusBackgroundColor;
    text-decoration: underline;
  }

  &.kuiTab-isSelected {
    cursor: default;
    color: $kuiColorSecondary;

    &:after {
      position: absolute;
      bottom: -1px;
      left: 0;
      content: ' ';
      width: 100%;
      height: $kuiBorderWidthThick;
      background-color: $kuiColorSecondary;
      animation: kuiTab $kuiAnimSpeedFast $kuiAnimSlightResistance;
    }
  }
}

  .kuiTab__content {
    display: block;
    transition: transform $kuiAnimSpeedFast $kuiAnimSlightBounce;
    transform: translateY(0);
  }


@keyframes kuiTab {
  0% {
    transform: scaleX(0);
  }
  100% {
    transform: scaleX(1);
  }
}
```